core/src/validator.rs

```diff
@@ -175,7 +179,32 @@ impl Validator {
             config.storage_slots_per_turn,
             bank.slots_per_segment(),
         );
-
+        if config.enable_ctrl_c_handler {
+            let bank_forks = bank_forks.clone();
+            let counter = Arc::new(AtomicUsize::new(0));
+            set_handler(move || {
+                if counter.fetch_add(1, Ordering::Relaxed) > 0 {
+                    warn!("Validator aborting on second ^C");
+                    std::process::exit(1);
+                }
+                let bank = bank_forks.read().unwrap().working_bank();
+                let _: Vec<_> = bank
```

Comment:
The point was that if the node currently has no stake it's ok to exit on the first ^C though.  I don't see that code anymore here.   It looks like we now collect all the stake but never check it for 0, and always exit on the second ^C